x-pack/plugins/search_indices/public/components/indices/details_page.tsx

```diff
@@ -208,20 +194,47 @@ export const SearchIndexDetailsPage = () => {
             data-test-subj="searchIndexDetailsHeader"
             pageTitle={index?.name}
             rightSideItems={[
-              <EuiFlexGroup>
+              <EuiFlexGroup gutterSize="none">
                 <EuiFlexItem>
-                  <EuiButtonEmpty
-                    href={docLinks.links.apiReference}
-                    target="_blank"
-                    iconType="documentation"
-                    data-test-subj="ApiReferenceDoc"
-                  >
-                    {i18n.translate('xpack.searchIndices.indexActionsMenu.apiReference.docLink', {
-                      defaultMessage: 'API Reference',
-                    })}
-                  </EuiButtonEmpty>
+                  {!isDocumentsExists ? (
+                    <EuiButtonEmpty
+                      href={docLinks.links.apiReference}
+                      target="_blank"
+                      isLoading={isDocumentsLoading}
+                      iconType="documentation"
+                      data-test-subj="ApiReferenceDoc"
+                    >
+                      <FormattedMessage
+                        id="xpack.searchIndices.indexAction.ApiReferenceButtonLabel"
+                        defaultMessage="{buttonLabel}"
+                        values={{
+                          buttonLabel: isDocumentsLoading ? 'Loading' : 'API Reference',
+                        }}
+                      />
```

Comment:
❗️Please avoid construction like this in the future as there is no translation created for the values.

Reply:
Question: There is a translation for the label  in formatted message id - `xpack.searchIndices.indexAction.ApiReferenceButtonLabel` . Do we still need separate translation for buttonLabel ?